src/kubernetes_multus.py

```diff
@@ -0,0 +1,276 @@
+            "config": {
+                "cniVersion": "0.3.1",
+                "type": "macvlan",
+                "ipam": {"type": "static"},
+                "capabilities": {"mac": True},
+            }
+        }
+    )
+
+    dict = asdict
+
+
+@dataclass
+class NetworkAnnotation:
+    """NetworkAnnotation."""
+
+    name: str
+    interface: str
+    ips: Optional[List] = None
+
+    dict = asdict
+
+
+class KubernetesMultusError(Exception):
+    """KubernetesMultusError."""
+
+    def __init__(self, message: str):
+        self.message = message
+        super().__init__(self.message)
+
+
+class KubernetesMultus:
+    """Class containing all the Kubernetes Multus specific calls."""
+
+    def __init__(self, namespace: str):
+        self.client = Client()
+        self.namespace = namespace
+
+    def network_attachment_definition_is_created(self, name: str) -> bool:
+        """Returns whether a NetworkAttachmentDefinitionResource is created.
+
+        Args:
+            name: NetworkAttachmentDefinition name
+
+        Returns:
+            bool: Whether the NetworkAttachmentDefinitionResource is created
+        """
+        try:
+            self.client.get(
+                res=NetworkAttachmentDefinitionResource,
+                name=name,
+                namespace=self.namespace,
+            )
+            logger.info(f"NetworkAttachmentDefinitionResource {name} already created")
+            return True
+        except ApiError as e:
+            if e.status.reason == "NotFound":
+                logger.info(f"NetworkAttachmentDefinitionResource {name} not yet created")
+                return False
+            else:
+                raise KubernetesMultusError(
+                    f"Unexpected outcome when retrieving network attachment definition {name}"
+                )
+        except httpx.HTTPStatusError as e:
+            if e.response.status_code == 404:
+                raise KubernetesMultusError(
+                    "NetworkAttachmentDefinitionResource resource not found. "
+                    "You may need to install Multus CNI."
+                )
+            else:
+                raise KubernetesMultusError(
+                    f"Unexpected outcome when retrieving network attachment definition {name}"
+                )
+
+    def create_network_attachment_definition(
+        self, network_attachment_definition: NetworkAttachmentDefinition
+    ) -> None:
+        """Creates a NetworkAttachmentDefinitionResource.
+
+        Args:
+            network_attachment_definition: NetworkAttachmentDefinition object
+        """
+        network_attachment_definition_obj = NetworkAttachmentDefinitionResource(
+            metadata=ObjectMeta(name=network_attachment_definition.name),
+            spec=network_attachment_definition.spec,
+        )
+        self.client.create(obj=network_attachment_definition_obj, namespace=self.namespace)  # type: ignore[call-overload]  # noqa: E501
```

Comment:
We should probably handle errors here

Reply:
If the charm fails at creating a network attachment definition it's not gonna work. so right now it would go to error state. What would be a better approach?

Reply:
Ok done, if there's other issues with the multus lib, let's discuss them in the PR for it [here](https://github.com/canonical/kubernetes-charm-libraries/pull/1)